pandas/core/arrays/sparse/array.py

```diff
@@ -1063,7 +1063,8 @@ def astype(self, dtype=None, copy=True):
         """
         dtype = self.dtype.update_dtype(dtype)
         subtype = dtype._subtype_with_str
-        sp_values = astype_nansafe(self.sp_values, subtype, copy=copy)
+        # TODO copy=False is broken for astype_nansafe with int -> float
+        sp_values = astype_nansafe(self.sp_values, subtype, copy=True)
```

Comment:
Opened https://github.com/pandas-dev/pandas/issues/34456 for this (and added link in the comment)